### ISSUE TITLE
[a11y]: Data table, keyboard tab selecting Search field even though Batch action bar is covering it

### ISSUE THREAD

**Issue by shixiedesign** [carbon-design-system/carbon#15813]
### Package

@carbon/react

### Package version

1.51.0

### Automated testing tool and ruleset

Manual testing

### Description

Currently, in data table with batch action, when batch action bar appears, it covers the search field. The problem is: while we cannot show both bars at the same time in the UI, it is possible to navigate to to both of them by using keyboard tab key. Which is a concerning accessibility issue because there is no visual indication on what is in focus in the UI (Read [Success Criterion 2.4.12 Focus Not Obscured](https://www.w3.org/TR/WCAG22/#focus-not-obscured-enhanced) and [Success Criterion 2.4.13 Focus Appearance](https://www.w3.org/TR/WCAG22/#focus-appearance)).

Credit goes to Rahul Sivananda for uncovering it. Slack discussion here: https://ibm-cio.slack.com/archives/C03HBU0F9J5/p1708590978617919

See screen recording:

https://github.com/carbon-design-system/carbon/assets/15144993/7f2aea16-e20e-4943-82f9-1f7624a735d0

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--playground

### Steps to reproduce

See screen recording attached above. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by shixiedesign** [comment]
Oops my mistake - this bug is unique to our LWC implementation. Closing.
